### PR TITLE
docs: rework docs site to have different sections

### PR DESCRIPTION
**Pull request by mimecuvalo**
This PR starts putting in place the high-level changes we want to make to the docs site.
- It makes separate sections for Reference and Examples and Community.
- Gets rid of the secondary sidebar and integrates it into the main sidebar.
- Groups the reference articles by type.
- Pulls in the examples alongside code and a live playground so people don't have to visit examples.tldraw.com separately.

Screenshot 2024-01-30 at 09 43 46

Again, this is the top-level changes and there's more to be done for the next PR(s):
  - create quick start page
  - clean up installation page
  - add accordion to Examples page prbly
  - put fun stuff in header (from footer)
  - landing page
  - something for landing page of API
  - search cmd-k and border
  - cleanup _sidebarReferenceContentLinks
  - external links _blank
  - address potential skew issue with code examples
  - have a link to other examples (next.js, etc.)

### Change Type

- [x] `documentation` — Changes to the documentation only[^2]

### Test Plan

1. Make sure examples work!

### Release Notes

- Rework our docs site to pull together the examples app and reference section more cohesively.

### CODE REVIEW

File: apps/docs/app/license/page.tsx
Comment (Taha-Hassan-Git) [inline apps/docs/app/license/page.tsx:1]: What was activeId used for before? I can't work it out.

File: apps/docs/components/ExampleCodeBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	files = {},
+	activeFile,
+}: {
+	articleId: string
+	activeFile: string
+	files: SandpackFiles
+}) {
+	const [isClientSide, setIsClientSide] = useState(false)
+	const { theme } = useTheme()
+	useEffect(() => setIsClientSide(true), [])
```
Comment (Taha-Hassan-Git) [inline apps/docs/components/ExampleCodeBlock.tsx:18]: Is this to get around some hydration weirdness?

File: apps/docs/app/cla/page.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ export default async function ClaPage() {
 
 	return (
 		<>
-			<Header activeId={null} />
```
Comment (Taha-Hassan-Git) [inline apps/docs/app/cla/page.tsx:11]: What was activeId doing before? I can't figure it out.

File: apps/examples/src/examples/canvas-events/CanvasEventsExample.tsx
Comment (Taha-Hassan-Git) [inline apps/examples/src/examples/canvas-events/CanvasEventsExample.tsx:1]: I don't think this file should be in here. Looks like one of my changes that leaked in from another branch, sorry!

**Review by Taha-Hassan-Git — APPROVED** [review]
Looks great!

File: apps/docs/components/ExampleCodeBlock.tsx (hunk 1 above)
Comment (mimecuvalo) [reply apps/docs/components/ExampleCodeBlock.tsx:18]: yeah, basically - i'll add a comment, now that you mention it 👍

File: apps/docs/app/cla/page.tsx (hunk 2 above)
Comment (mimecuvalo) [reply apps/docs/app/cla/page.tsx:11]: nothing! :P

Comment (mimecuvalo) [reply apps/docs/app/cla/page.tsx:11]: it _was_ being used in another page (maybe ArticleDocsPage?) but it was copy-pasted over i think needlessly

**Pull request merged**